runtime/src/snapshot_package.rs

```diff
@@ -23,14 +22,9 @@ use {
     tempfile::TempDir,
 };
 
-/// The sender side of the AccountsPackage channel, used by AccountsBackgroundService
-pub type AccountsPackageSender = Sender<AccountsPackage>;
-
-/// The receiver side of the AccountsPackage channel, used by AccountsHashVerifier
-pub type AccountsPackageReceiver = Receiver<AccountsPackage>;
-
-/// The error type when sending an AccountsPackage over the channel fails
-pub type AccountsPackageSendError = SendError<AccountsPackage>;
+/// The PendingAccountsPackage passes an AccountsPackage from AccountsBackgroundService to
+/// AccountsHashVerifier for hashing
+pub type PendingAccountsPackage = Arc<Mutex<Option<AccountsPackage>>>;
```

Comment:
nit: rename to `PendingAccountsSnapshotPackage`

Reply:
There's a `SnapshotPackage`, and corresponding `PendingSnapshotPackage`, for AHV -> SnapshotPackagerService. I feel like renaming to `PendingAccountsSnapshotPackage` would be confusing. Maybe there's a different name that'll work better?

Reply:
ah dang thats annoying. Yeah maybe there's some name that better captures the difference:

1. What is the difference between `PendingSnapshotPackage` and `PendingAccountsSnapshotPackage`
2. How is the former converted to the later?

Reply:
`AccountsPackage` goes into the `PendingAccountsPackage` for passing between ABS and AHV.
`SnapshotPackage` goes into the `PendingSnapshotPackage` for passing between AHV and SPS.

The `AccountsPackage` has the bits necessary for (1) accounts hash calculation (work in progress to move this from ABS to AHV), (2) bits needed for hash verification on CRDS, and testing/validation, (3) snapshot archives (if this package will be archived).

After the `AccountsPackage` does all the AHV-specific stuff, it may or may not become a snapshot archive. I.e. if the snapshot archive interval is not the same as the accounts hash interval, then some `AccountsPackage`s will not become `SnapshotPackage`s. If it is going to be a snapshot archive, then it is converted into a `SnapshotPackage` to have just the bits left to make the snapshot archive.

I'm not sure if there's a clear enough delineation to give an obvious better name. Maybe it can be updated later if a better name is discovered?

Reply:
sounds good to me!